src/main/java/org/sonarlint/intellij/config/global/SonarQubeServerEditorPanel.java

```diff
@@ -0,0 +1,235 @@
+
+import static java.awt.GridBagConstraints.NONE;
+import static java.awt.GridBagConstraints.WEST;
+
+public class SonarQubeServerEditorPanel implements Disposable {
+  private JBLabel urlLabel;
+  private JBTextField urlText;
+
+  private JBLabel nameLabel;
+  private JBTextField nameText;
+
+  private JBLabel idLabel;
+  private JBTextField idText;
+
+  private JBTextField loginText;
+  private JBLabel loginLabel;
+
+  private JBPasswordField passwordText;
+  private JBLabel passwordLabel;
+
+  private JBCheckBox enableProxy;
+  private JButton proxySettingsButton;
+
+  protected JButton testButton;
+  private final Consumer<SonarQubeServer> changeListener;
+
+  private SonarQubeServer server;
+
+  public SonarQubeServerEditorPanel(Consumer<SonarQubeServer> changeListener, SonarQubeServer server) {
+    this.changeListener = changeListener;
+    this.server = server;
+  }
+
+  public JComponent create() {
+    nameLabel = new JBLabel("Name:", SwingConstants.RIGHT);
+    nameLabel.setDisplayedMnemonic('N');
+    nameText = new JBTextField();
+    nameText.setText(server.getName());
+    nameText.getEmptyText().setText("");
+    nameLabel.setLabelFor(nameText);
+
+    idLabel = new JBLabel("Server ID:", SwingConstants.RIGHT);
+    idText = new JBTextField();
+    idText.setText(server.getServerId());
+    idText.getEmptyText().setText("");
+    idLabel.setLabelFor(idText);
+
+    urlLabel = new JBLabel("Server URL:", SwingConstants.RIGHT);
+    urlLabel.setDisplayedMnemonic('U');
+    urlText = new JBTextField();
+    urlText.setText(server.getHostUrl());
+    urlText.getEmptyText().setText("Example: http://localhost:9000");
+    urlLabel.setLabelFor(urlText);
+
+    loginLabel = new JBLabel("Login or token:", SwingConstants.RIGHT);
+    loginLabel.setDisplayedMnemonic('L');
+    loginText = new JBTextField();
+    loginText.setText(server.getLogin());
+    loginText.getEmptyText().setText("");
+    loginLabel.setLabelFor(loginText);
+
+    passwordLabel = new JBLabel("Password:", SwingConstants.RIGHT);
+    passwordText = new JBPasswordField();
+    passwordText.setText(server.getPassword());
+    passwordText.getEmptyText().setText("");
+    passwordLabel.setLabelFor(passwordText);
+
+    proxySettingsButton = new JButton("Proxy settings");
+    enableProxy = new JBCheckBox("Use proxy", server.enableProxy());
+    enableProxy.setMnemonic('y');
+
+    enableProxy.setEnabled(HttpConfigurable.getInstance().USE_HTTP_PROXY);
+
+    testButton = new JButton("Test connection");
+    testButton.setHorizontalAlignment(SwingConstants.RIGHT);
+    testButton.setAlignmentX(Component.RIGHT_ALIGNMENT);
+    testButton.addActionListener(new ActionListener() {
+      public void actionPerformed(ActionEvent e) {
+        testConnection();
+      }
+    });
+
+    final JPanel p = new JPanel(new GridBagLayout());
+    addForm(p);
+
+    proxySettingsButton.addActionListener(new ActionListener() {
+      public void actionPerformed(ActionEvent e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add the "@Override" annotation above this method signature [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1161)